From 31d99c4c9f14e571ca0e22c5862dd5b6a871e2dd Mon Sep 17 00:00:00 2001
From: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
Date: Fri, 5 Aug 2011 15:32:23 +0400
Subject: [PATCH] pcmcia/pxa2xx_sharpsl: retain ops structure on collie

The pxa2xx_sharpsl driver part is also used on collie, which (as
a StrongARM board) has different expectations for PCMCIA drivers.
So, on collie place sharpsl_pcmcia_ops in .data section rather than
in __initdata.

Signed-off-by: Dmitry Eremin-Solenikov <dbaryshkov@gmail.com>
---
 drivers/pcmcia/pxa2xx_sharpsl.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/pcmcia/pxa2xx_sharpsl.c b/drivers/pcmcia/pxa2xx_sharpsl.c
index 81af2b3..1b5b749 100644
--- a/drivers/pcmcia/pxa2xx_sharpsl.c
+++ b/drivers/pcmcia/pxa2xx_sharpsl.c
@@ -222,7 +222,11 @@ static void sharpsl_pcmcia_socket_suspend(struct soc_pcmcia_socket *skt)
 	sharpsl_pcmcia_init_reset(skt);
 }
 
+#ifdef CONFIG_SA1100_COLLIE
+static struct pcmcia_low_level sharpsl_pcmcia_ops = {
+#else
 static struct pcmcia_low_level sharpsl_pcmcia_ops __initdata = {
+#endif
 	.owner                  = THIS_MODULE,
 	.hw_init                = sharpsl_pcmcia_hw_init,
 	.hw_shutdown            = sharpsl_pcmcia_hw_shutdown,
-- 
1.7.2.5

